src/main/webapp/dashboard-manager.js

```diff
@@ -0,0 +1,148 @@
+          .then(response => response.json())
+          .then(data => {
+              for (let [key, val] of Object.entries(data)) {
+                this.data.addRow([key, val]);
+              }
+      }));
+    } else {
+      // Setup data for standard table view
+      this.data.addColumn('string', 'Serial Number');
+      this.data.addColumn('string', 'Status');
+      this.data.addColumn('string', 'Asset ID');
+      this.data.addColumn('string', 'User');
+      this.data.addColumn('string', 'Location');
+
+      /* TODO: once the server can send a list of devices, we can get real data here.
+      await (fetch('/devices')
+          .then(response => response.json())
+          .then(deviceJsons => {
+              for (let device of deviceJsons) {
+                this.data.addRow([
+                    device.serialNumber,
+                    device.status,
+                    device.assetId,
+                    device.user,
+                    device.location]);
+              }
+      }));
+      */
+      this.data.addRow(['SN12345', 'Provisioned', '1e76c3', 'James', 'Texas']);
+      this.data.addRow(['SN54321', 'Provisioned', 'a9f27d', 'Justin', 'Alaska']);
+    }
+
+    this.draw();
+  }
+
+  isAggregating() {
+    // If no aggregation tags are selected, we are not currently aggregating
+    return this.aggregationSelector.getState().selectedValues.length != 0;
+  }
+
+  draw() {
+    this.dashboard.draw(this.data);
+
+    this.table.setDataTable(this.data);
+
+    this.table.draw();
+    this.aggregationSelector.draw();
+
+    if (this.isAggregating()) {
+      this.pieChart.setDataTable(this.data);
+      this.pieChart.draw();
+    }
+  }
+};
+
+
+
+
+function createNewTable() {
+  return new google.visualization.ChartWrapper({
+      'chartType': 'Table',
+      'containerId': 'table-container',
+      'options': {
+          'title': 'Sample Table',
+          'width': '100%',
+          'allowHtml': 'true',
+          'cssClassNames': {
+              'headerRow': 'table-header-row',
+              'tableRow': 'table-row',
+              'oddTableRow': 'table-row',
+              'tableCell': 'table-cell'
+          }
+      }
+  });
+}
+
+function createNewPieChart() {
+  return new google.visualization.ChartWrapper({
+      'chartType': 'PieChart',
+      'containerId': 'piechart-container',
+      'options': {
+          'title': 'Devices by Location',
+          'legend': 'none',
+          'width': '100%',
+          'height': '50%',
+          'chartArea': {
+              'left': '5', 'width': '100%', 'height': '95%'
+          }
+      },
+      'view': {'columns': [0, 1]}
+  });
+}
+
+function createNewAggregationSelector() {
+  const aggregationOptions = new google.visualization.DataTable();
+  aggregationOptions.addColumn('string', 'Aggregation Options');
+  aggregationOptions.addRows([['annotatedAssetId'], ['annotatedLocation'], ['annotatedUser']]);
+
+  return new google.visualization.ControlWrapper({
```

Comment:
Seems like we only support aggregate by one field at the moment. But I saw we can add multiple fields in UI during today's demo. Is there anything we can do to disable adding multiple fields until we support the functionality? 

If it's a big change I'm okay with leaving a todo for now.

Reply:
Thankfully it was a simple change, the Google Charts API CategoryFilter class has an option called ["ui.allowMultiple"](https://developers.google.com/chart/interactive/docs/gallery/controls#options) that I could set to false.